src/ReasonReactRouter.re

```diff
@@ -133,14 +144,42 @@ type url = {
   search: string,
 };
 type watcherID = unit => unit;
-let url = () => {path: path(), hash: hash(), search: search()};
+let urlFromWindow = () => {
+  path: pathFromWindow(),
+  hash: hashFromWindow(),
+  search: searchFromWindow(),
+};
+let url = (~pathname=?, ~search=?, ~hash=?, ()) => {
+  path:
+    pathFromString(
+      switch (pathname) {
+      | Some(pathname) => pathname
+      | None => "/"
+      },
+    ),
+  search:
+    searchFromString(
```

Comment:
Same as below, this is weird because URL in JS provides query without the leading `?` so you have to add it yourself.